src/System.Numerics.Tensors/src/System/Numerics/Tensors/ArrayUtilities.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Numerics.Tensors
+{
+    internal static class ArrayUtilities
+    {
+        public const int StackallocMax = 16;
```

Comment:
This appears to always be used with Int32s, which means the limit being imposed here is only 64 bytes.  We can generally afford much more stack space than that, if we wanted to increase this.